mm2src/mm2_main/src/rpc/dispatcher/dispatcher_legacy.rs

```diff
@@ -77,8 +77,8 @@ pub fn dispatcher(req: Json, ctx: MmArc) -> DispatcherRes {
         "get_gossip_peer_topics" => hyres(get_gossip_peer_topics(ctx)),
         "get_gossip_topic_peers" => hyres(get_gossip_topic_peers(ctx)),
         "get_my_peer_id" => hyres(get_my_peer_id(ctx)),
-        "get_peers_info" => hyres(get_peers_info(ctx)),
         "get_relay_mesh" => hyres(get_relay_mesh(ctx)),
+        "get_seeds_info" => hyres(get_seeds_info(ctx)),
```

Comment:
what if `"get_seeds_info" | "get_peers_info" => ...`

Reply:
Good idea, but IMO this change should break current usage to inform people that we don't return the all connected peers. The current name and its documentation is misleading.

Reply:
What about if this is used from a seed node, does it return connected seeds info only? if so, then renaming it is fine, if not, then `get_peers_info` is actually accurate since light nodes connect to seed nodes only and this returns the connected peers, it can be called `get_connected_peers_info` in such case.

I tried the below test where bob is a seednode and eve/alice are both light nodes that connect to bob
```rust
#[test]
fn test_connected_peers() {
    let coins = json!([]);

    let mm_bob = MarketMakerIt::start(
        json!({
            "gui": "nogui",
            "netid": 9998,
            "passphrase": "bob passphrase",
            "coins": coins,
            "rpc_password": "pass",
            "i_am_seed": true,
        }),
        "pass".into(),
        None,
    )
    .unwrap();
    let (_bob_dump_log, _bob_dump_dashboard) = mm_bob.mm_dump();
    log!("Bob log path: {}", mm_bob.log_path.display());

    let mm_eve = MarketMakerIt::start(
        json!({
            "gui": "nogui",
            "netid": 9998,
            "passphrase": "eve passphrase",
            "coins": coins,
            "rpc_password": "pass",
            "seednodes": [mm_bob.ip.to_string()],
        }),
        "pass".into(),
        None,
    )
    .unwrap();
    let (_eve_dump_log, _eve_dump_dashboard) = mm_eve.mm_dump();
    log!("Eve log path: {}", mm_eve.log_path.display());

    let mm_alice = MarketMakerIt::start(
        json!({
            "gui": "nogui",
            "netid": 9998,
            "passphrase": "alice passphrase",
            "coins": coins,
            "seednodes": [mm_bob.ip.to_string()],
            "rpc_password": "pass",
        }),
        "pass".into(),
        None,
    )
    .unwrap();

    let (_alice_dump_log, _alice_dump_dashboard) = mm_alice.mm_dump();
    log!("Alice log path: {}", mm_alice.log_path.display());

    let bob_connected_peers = block_on(mm_bob.rpc(&json!({
        "userpass": mm_bob.userpass,
        "method": "get_seeds_info",
    })))
    .unwrap().1;
    let bob_connected_peers: Json = serde_json::from_str(&bob_connected_peers).unwrap();
    let bob_connected_peers = bob_connected_peers.get("result").unwrap();

    let eve_connected_peers = block_on(mm_eve.rpc(&json!({
        "userpass": mm_eve.userpass,
        "method": "get_seeds_info",
    })))
    .unwrap().1;
    let eve_connected_peers: Json = serde_json::from_str(&eve_connected_peers).unwrap();
    let eve_connected_peers = eve_connected_peers.get("result").unwrap();

    let alice_connected_peers = block_on(mm_alice.rpc(&json!({
        "userpass": mm_alice.userpass,
        "method": "get_seeds_info",
    })))
        .unwrap().1;
    let alice_connected_peers: Json = serde_json::from_str(&alice_connected_peers).unwrap();
    let alice_connected_peers = alice_connected_peers.get("result").unwrap();

    let bob_peer_id = block_on(mm_bob.rpc(&json!({
        "userpass": mm_bob.userpass,
        "method": "get_my_peer_id",
    })))
        .unwrap().1;
    let bob_peer_id: Json = serde_json::from_str(&bob_peer_id).unwrap();
    let bob_peer_id = bob_peer_id.get("result").unwrap();

    let alice_peer_id = block_on(mm_alice.rpc(&json!({
        "userpass": mm_alice.userpass,
        "method": "get_my_peer_id",
    })))
        .unwrap().1;
    let alice_peer_id: Json = serde_json::from_str(&alice_peer_id).unwrap();
    let alice_peer_id = alice_peer_id.get("result").unwrap();

    let eve_peer_id = block_on(mm_eve.rpc(&json!({
        "userpass": mm_eve.userpass,
        "method": "get_my_peer_id",
    })))
    .unwrap().1;
    let eve_peer_id: Json = serde_json::from_str(&eve_peer_id).unwrap();
    let eve_peer_id = eve_peer_id.get("result").unwrap();

    let bob_ip = mm_bob.ip.to_string();
    let alice_ip = mm_alice.ip.to_string();
    let eve_ip = mm_eve.ip.to_string();

    println!("bob_connected_peers: {}", bob_connected_peers);
    println!("eve_connected_peers: {}", eve_connected_peers);
    println!("alice_connected_peers: {}", alice_connected_peers);
    println!("bob_peer_id: {}", bob_peer_id);
    println!("alice_peer_id: {}", alice_peer_id);
    println!("eve_peer_id: {}", eve_peer_id);
    println!("bob_ip: {}", bob_ip);
    println!("alice_ip: {}", alice_ip);
    println!("eve_ip: {}", eve_ip);

    block_on(mm_bob.stop()).unwrap();
    block_on(mm_alice.stop()).unwrap();
    block_on(mm_eve.stop()).unwrap();
}
```

This is the printed output
```
bob_connected_peers: {"12D3KooWLVMh8iyJe8KJjWAj17HS9b3ZZaXtcg4rPM9SVsxuEKnY":["/ip4/127.0.0.169/tcp/57106"],"12D3KooWMNPaCEAT2UdFS1oZw51eK7tcjLZQo3DynzjqmYi7bBEj":["/ip4/127.0.0.169/tcp/57108"]}
eve_connected_peers: {"12D3KooWA7XvEuVtvK8r7Gn6cN48QewFLPHn3rSWF9zhWgASKs7G":["/ip4/127.0.0.169/tcp/47771"]}
alice_connected_peers: {"12D3KooWA7XvEuVtvK8r7Gn6cN48QewFLPHn3rSWF9zhWgASKs7G":["/ip4/127.0.0.169/tcp/47771"]}
bob_peer_id: "12D3KooWA7XvEuVtvK8r7Gn6cN48QewFLPHn3rSWF9zhWgASKs7G"
alice_peer_id: "12D3KooWMNPaCEAT2UdFS1oZw51eK7tcjLZQo3DynzjqmYi7bBEj"
eve_peer_id: "12D3KooWLVMh8iyJe8KJjWAj17HS9b3ZZaXtcg4rPM9SVsxuEKnY"
bob_ip: 127.0.0.169
alice_ip: 127.0.0.165
eve_ip: 127.0.0.167
```

So the seed node response of `get_seeds_info` returns both alice's and eve's peer_id but the ip is bob's ip with the port used from bob's side for both alice and eve. So this actually returns peers info not seeds info.

Reply:
> What about if this is used from a seed node, does it return connected seeds info only?

No, for seed nodes this can return non-seed nodes as well. But still it may not include all connected peers. We might want to consider renaming this RPC to something more descriptive like "get_direct_connected_peers".

Reply:
> We might want to consider renaming this RPC to something more descriptive like "get_direct_connected_peers".

Agree, `get_direct_connected_peers` or `get_directly_connected_peers` are good descriptive names.

Reply:
Done that.